tests/integration/test_kfp_functional_v2.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Functional tests for kfp-operators with the KFP SDK v2."""
+import logging
+import time
+from pathlib import Path
+
+from helpers.bundle_mgmt import render_bundle, deploy_bundle
+from helpers.k8s_resources import apply_manifests, fetch_response
+from helpers.localize_bundle import get_resources_from_charm_file
+from kfp_globals import (
+    CHARM_PATH_TEMPLATE,
+    KFP_CHARMS,
+    KUBEFLOW_PROFILE_NAMESPACE,
+    SAMPLE_PIPELINE,
+    SAMPLE_PIPELINE_NAME,
+    SAMPLE_VIEWER,
+)
+
+import kfp
+import lightkube
+import pytest
+import tenacity
+from lightkube import codecs
+from lightkube.generic_resource import create_namespaced_resource
+from lightkube.resources.apps_v1 import Deployment
+from pytest_operator.plugin import OpsTest
+
+KFP_SDK_VERSION = "v2"
+log = logging.getLogger(__name__)
+
+
+# ---- KFP SDK V2 fixtures
+@pytest.fixture(scope="function")
+def upload_and_clean_pipeline_v2(kfp_client: kfp.Client):
+    """Upload an arbitrary v2 pipeline and remove after test case execution."""
+    pipeline_upload_response = kfp_client.pipeline_uploads.upload_pipeline(
+        uploadfile=SAMPLE_PIPELINE["v2"], name=SAMPLE_PIPELINE_NAME
```

Comment:
Same here, you could use `KFP_SDK_VERSION` here, right?
```suggestion
        uploadfile=SAMPLE_PIPELINE[KFP_SDK_VERSION], name=SAMPLE_PIPELINE_NAME
```